common/app/templates/inlineJS/blocking/config.scala.js

```diff
@@ -4,19 +4,21 @@
 @import views.support.{JavaScriptPage, CamelCase}
 @import play.api.libs.json.Json
 
+var isModernBrowser =
+    "querySelector" in document
+    && "addEventListener" in window
+    && "localStorage" in window
+    && "sessionStorage" in window
+    && "bind" in Function
+    && (
+        ("XMLHttpRequest" in window && "withCredentials" in new XMLHttpRequest())
+        || "XDomainRequest" in window
+    );
+
 window.guardian = {
-    isModernBrowser: (
-        window.shouldEnhance
-            && "querySelector" in document
-            && "addEventListener" in window
-            && "localStorage" in window
-            && "sessionStorage" in window
-            && "bind" in Function
-            && (
-                ("XMLHttpRequest" in window && "withCredentials" in new XMLHttpRequest())
-                    || "XDomainRequest" in window
-            )
-    ),
+    isModernBrowser : isModernBrowser,
+    isEnhanced:
+        window.shouldEnhance && isModernBrowser,
```

Comment:
I see what you mean. I guess it's pretty easy to just make `shouldEnhance.scala.js` decorate the isEnhanced field. This would mean we wouldn't pollute the namespace with temporary values, and be a fairly minor (i.e. low risk) change overall.